### ISSUE TITLE
FR: amp-list/amp-bind fetching status attribute

### ISSUE THREAD

**Issue by ruborg** [ampproject/amphtml#10802]
## What's the issue?

When data is submitted with forms/amp-form, useful status classes are applied to the form element:
* class: `amp-form-submitting`
* class: `amp-form-submit-success`

These classes allow the developer to do nice UX stuff like displaying a loading spinner, or some other feedback for the user that something is happening.

With `amp-bind` and `amp-list`, we can produce similar behaviour to that with `amp-form`, but without the restriction of the results being confined to the `submit-success` container. `amp-list`/`amp-bind` is more flexible and powerful.

It would be really great to add a class or attribute to the` amp-list` element when a new fetch occurs, so that the UX for user could be improved by indicating a background behaviour.

It could work like this:
**If**
1. the `src` attribute has been bound, AND
2. `src` is updated AND
3. new fetch occurs

**Then**
add class (or attribute) `fetching`.
When fetch completes or fails, then add `fetch-success` or `fetch-failed`

**Comment by aghassemi** [comment]
/cc @ericlindley-g related to the discussion we had on Thursday about `sync` and `amp-animation`. 
If we add state classes to `amp-list`, content jumps/abuse becomes hard to control but if we tie it with `amp-animation` it enables certain UX possibilities without the risks.

**Comment by ericlindley-g** [comment]
Thanks for looping me in — General FR sounds great (could be super-useful for infinite scroll, when that's available). After chatting offline w/ @aghassemi:

Given that amp-list updates only at load time (when resize is restricted) or when scrolled into viewport (current behavior—resize only when possible without content jumps) or when triggered by meaningful user interaction (via amp-bind), we should be able to avoid unexpected content jumps. 

This is because we can either constrain the loading state to the existing size (if meaningful user interaction hasn't happened), or have a resize triggered by user interaction (amp-bind). I would generally think of this state like '<placeholder>' or '<fallback>', where arbitrary AMPHTML can be used to define an intermediate placeholder (though amp-animation could be used at some point as well)

**Comment by aghassemi** [comment]
deduping with #12212
